armi/reactor/blueprints/assemblyBlueprint.py

```diff
@@ -153,17 +192,67 @@ def _constructAssembly(self, cs, blueprint):
 
         return a
 
-    def _createBlock(self, cs, blueprint, bDesign, axialIndex):
+    def _compileAssemblyMaterialInput(self):
+        """
+        Combine the blockwise and componentwise material inputs for the entire
+        assembly into a single dict `assemblyMaterialInput`. The combined dict
+        has the structure:
+        ```
+        {"": {blockwise_mod_name1: blockwise_mod_list1}, ..., {blockwise_mod_nameN: blockwise_mod_listN},
+         comp_name1: {componentwise_mod_name1: componentwise_mod_list1}, ..., {componentwise_mod_nameN: componentwise_mod_listN},
+         .
+         .
+         .
+         comp_nameN: {componentwise_mod_name1: componentwise_mod_list1}, ..., {componentwise_mod_nameN: componentwise_mod_listN}}
+         ```
+
+        For each `mod_list`, there are expected to be as many entries as there are
+        blocks in the assembly. This may be relaxed in the future by allowing a
+        single value to apply to all blocks without needing to write out a whole list.
+
+        To maintain backwards compatibility, we first assumed that there might be
+        blockwise matMods on the base object, then we add in any blockwise matMods
+        that are explicitly denoted using the `blockwise` attribute.
+        """
+        assemblyMaterialInput = {}
+
+        if self.materialModifications != None:
+            assemblyMaterialInput[""] = {}
```

Comment:
I recommend defining a special module-level constant like `BLOCKWISE = "_BLOCKWISE"` and using that instead of `""` as the blockwise key. That will add some additional useful context.

Reply:
I struggled with coming up with a good key name for the blockwise material modifications, because I didn't want to use a name that might possibly conflict with a name that the user might have for a component. 

Like, if the user were to name a component `_BLOCKWISE` for whatever odd reason, then any material modifications to that component would be overwritten internally. I know it is unlikely that a user might use that exact name, but still... Trying to avoid conflicts here.

I figured the safest way would be to not use a real word at all, hence the use of `""`. I figured that yamlize wouldn't allow an empty string to be used as an attribute name, but now that I actually checked, I find out I was wrong. So I'll change it to something more clear.

Reply:
I changed the key.